### ISSUE TITLE
Feature Request: Allow Sidekiq::Batch::LINGER value to be customized

### ISSUE THREAD

**Issue by hasghari** [mperham/sidekiq#4772]
Currently in Sidekiq Pro, the value for `Sidekiq::Batch::LINGER` is set to one day. The only way to change this value is to reassign the constant which produces a warning. Ideally this should be configurable globally and per batch instance.

**Comment by mperham** [comment]
Why do you want to change linger?

**Comment by hasghari** [comment]
We create a lot of batches that store data in the options hash for the callback. We want to be able to reclaim the memory once the batch has succeeded. The comments in the code claim this should be possible:
```ruby
# Controls how long a batch record "lingers" in Redis before expiring.
# This allows APIs like Status#poll to check batch status even after
# the batch succeeds and is no longer needed.  You can lower this
# constant if you create lots of batches, want to reclaim the memory
# and don't use polling.
LINGER = ONE_DAY
```

**Comment by mperham** [comment]
Thanks, that's really helpful. I've added this API which will be available in Pro 5.2.1, coming later this month:

```ruby
batch = Sidekiq::Batch.new
batch.linger = 600 # 10 minutes
batch.jobs do
   ...
end
```

**Comment by hasghari** [comment]
Awesome, thank you!
